File: packages/angular/src/app/core/models/user/user-dto.ts

export interface IUserDto {
  id: string;
  userName: string;
  email: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  imageUrl: string;
  // token: string;
}

export interface IUserDtoDetails extends IUserDto {
  roles: string[];
}

export interface ICreateUserDto {
  userName: string;
  email: string;
  password: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
}

export interface IUpdateUserDto {
  email: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  roles: string[];
  imageUrl: string;
}

export interface ILoginDto {
  userName: string;
  password: string;
}

File: api/CuentaClaraAPI/CuentaClara.Application/DTOs/UserDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CuentaClara.Application.DTOs
{
    public class UserDto
    {
        public string Id { get; set; } = null!;
        public string UserName { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string PhoneNumber { get; set; } = null!;
        public string ImageUrl { get; set; } = null!;

        //public string? Token { get; set; }
    }
}
